Tools/Incinerator/GLTFLoader.cpp

```diff
@@ -171,6 +194,68 @@ bool ParseAndIncinerate(const uint8_t* gltfData,
 		std::ofstream outputFile(destinationFolder + meshName, std::ios::binary | std::ios::out | std::ios::trunc);
 		outputFile.write(reinterpret_cast<const char*>(memstream.GetData()), memstream.GetDataSize());
 	}
+
+	for (const auto& materialIndex : materialToIncinerate)
+	{
+		Zmey::Graphics::MaterialDataHeader dataHeader;
+		// Default data
+		dataHeader.BaseColorFactor = Zmey::Color(1.0f, 1.0f, 1.0f, 1.0f);
+		dataHeader.BaseColorTextureOffset = 0; // no texture
+		dataHeader.BaseColorTextureSize = 0;
+
+		uint32_t baseColorTextureIndex = -1;
+
+		const auto& material = FindMaterial(gltf, materialIndex);
+		if (material.find("pbrMetallicRoughness") != material.end())
+		{
+			auto pbr = material["pbrMetallicRoughness"];
+			assert(pbr.is_object());
+			for (const auto& member : pbr.get<nlohmann::json::object_t>())

```

Comment:
maybe we can split this function ?

Reply:
Yes definitely. However not right now. I want to implement a little more stuff inside the loader as to see how everything is going to be structured. After that I will split it into more managable functions. This will be done while I m doing the skeletons which will again change a lot of stuff inside the loader.